webapp/lib/controller.dart

```diff
@@ -426,16 +425,26 @@ void command(UIAction action, Data data) {
         );
       } else if (data is TranslationData) {
         TranslationData messageTranslation = data;
-        activeConversation.messages[messageTranslation.messageIndex].translation = messageTranslation.translationText;
-        platform.updateConversationMessages(activeConversation);
+        var conversation = activeConversation;
+        var message = conversation.messages[messageTranslation.messageIndex];
+        SaveTextAction.textChange(
+          "${conversation.docId}.message-${messageTranslation.messageIndex}.translation",
+          messageTranslation.translationText,
+          (newText) {
+            return platform.setMessageTranslation(conversation, message, newText);
+          },
+        );
       }
       break;
-    case UIAction.updateNote:
-      SaveTextAction.textChange(
-        "${activeConversation.docId}.notes",
-        (data as NoteData).noteText,
-        (newText) => platform.updateNotes(activeConversation, newText),
-      );
+    case UIAction.updateNote: {
+        // inside block to ensure conversation is a local/captured var
```

Comment:
why is this an issue? and if it is, would you need to have an inside block for the above code as well?

Reply:
`activeConversation` is a global which can change before the `(newText) => ...` closure executes. Thus I assign `activeConversation` to a local var `conversation` to capture the value.

On hindsight, the `{ }` are not necessary and can be removed. I'll tackle that in a subsequent PR.

Reply:
Fixed in https://github.com/larksystems/nook/pull/222